src/calendar.pipe.ts

```diff
@@ -64,20 +64,23 @@ export class CalendarPipe implements PipeTransform, OnDestroy {
     this.midnightSub.unsubscribe();
   }
 
-  private static initTimer() {
+  private static initTimer(ngZone: NgZone, cdRef: ChangeDetectorRef) {
     // initialize the timer
     if (!CalendarPipe.midnight) {
       CalendarPipe.midnight = new EventEmitter<Date>();
       if (typeof window !== 'undefined') {
         let timeToUpdate = CalendarPipe._getMillisecondsUntilUpdate();
-        CalendarPipe.timer = window.setTimeout(() => {
-          // emit the current date
-          CalendarPipe.midnight.emit(new Date());
-
-          // refresh the timer
-          CalendarPipe.removeTimer();
-          CalendarPipe.initTimer();
-        }, timeToUpdate);
+        CalendarPipe.timer = ngZone.runOutsideAngular(() => {
+          window.setTimeout(() => {
+            // emit the current date
+            CalendarPipe.midnight.emit(new Date());
+
+            // refresh the timer
+            CalendarPipe.removeTimer();
+            CalendarPipe.initTimer(ngZone, cdRef);
+            ngZone.run(() => cdRef.markForCheck());
+          }, timeToUpdate);
```

Comment:
We don't really need to run the `markForCheck()` here - we emit an event on the `midnight` emitter, and the constructor subscribes and runs the change detector. Since there can be several change detector refs, it makes sense to keep this call inside the constructor.

Also, I think we can remove the `runOutsideAngular()` from the constructor itself - it seems to be unnecessary left over from times when the timer was created directly inside the constructor.